embedder/text_embedder.py

```diff
@@ -62,10 +74,10 @@ def embed_content(self, text, clean_text=True, use_single_precision=True):
         if clean_text:
             text = self.clean_text(text)        
         try:
-            response = openai.Embedding.create(input=text, engine=self.AZURE_OPENAI_EMBEDDING_DEPLOYMENT)
-            embedding = [round(x, embedding_precision) for x in response['data'][0]['embedding']] # type: ignore
+            response = client.embeddings.create(input=text, model=self.AZURE_OPENAI_EMBEDDING_DEPLOYMENT)
+            embedding = [round(x, embedding_precision) for x in response.data[0].embedding] # type: ignore
             return embedding            
-        except openai.error.RateLimitError as e:
+        except Exception as e:
```

Comment:
@namtran6701 you might consider handle non-rate-limit exceptions differently.  because if the error is something else (e.g., invalid model name), you might be need to differentiate between error types